Add loaded_from_cache flag to EsiResponse

michaelcooke opened the pull request:
Addresses #22.

The approach I took sets the flag right before the EsiResponse is returned in Eseye::invoke() and should work regardless of the cache implementation used.

I would include tests for this but I don't know where to start on that front. That being said, I've tested it by hand and it seems to work well enough.

leonjza commented:
Thanks for this!

I think there is a small misunderstanding for the requirement though. For #22, the idea is when a response is returned from the cache, a boolean should be set on the returned `EsiResponse` object as a property or something indicating that this response was loaded from the cache. This is something I think we can implement as a getter/setter on [EsiResponse](https://github.com/eveseat/eseye/blob/master/src/Containers/EsiResponse.php), and can be set in [this](https://github.com/eveseat/eseye/blob/e9e1763c3b7bca3b02bbd49540efd5e6f88a587b/src/Eseye.php#L242-L246) conditional. This will serve as a relatively cheap flag indicating that nothing has changed since the last request (something super useful for SeAT!).

As a last note, I think you may have accidentally omitted the `{}` braces in the following [lines](https://github.com/michaelcooke/eseye/blob/5eec93565b6629993638357e071040f2f08dbec7/src/Eseye.php#L245-L249). One can only omit the braces if there is a single following statement. Multiline statements need to be enclosed in curly braces. :)

leonjza commented:
I realised I haven't had coffee yet. The changes to `EsiResponse` look fine, it just the location of where we flag it as cached that needs to move up ;)

michaelcooke commented:
I renamed the flag and its accompanying setter to the more obvious loaded_from_cache and setLoadedFromCache(), respectively.

I've also moved the point in which the cached flag is set to just before the EsiResponse is returned instead of setting it in the cache with a fresh request.

Regarding the if statement brackets or lack thereof, I tried to be as conforming as possible to the style and conventions you follow for the package -- I didn't know why you wouldn't have brackets in the first place and I went with it. :P

That being said, you cleared that up and all that code is gone anyway, but let me know if there are any other style preferences that both StyleCI and I missed.

leonjza commented:
Nice thank you! With the property defaulting to `false` I would have a setter `setLoadedFromCache()` not accept an argument as we can just default that to `true` when called ;)

I'll test this in a bit, thanks again.

michaelcooke commented:
Good catch; I've went ahead and updated that.
